util/package.go

```diff
@@ -316,9 +315,32 @@ func (p *Package) Validate() error {
 		}
 	}
 
+	err = p.validateVersionConsistency()
+	if err != nil {
+		return errors.Wrap(err, "version in manifest file is not consistent with path")
+	}
+
 	return p.ValidateDatasets()
 }
 
+func (p *Package) validateVersionConsistency() error {
+	versionPackage, err := semver.Parse(p.Version)
```

Comment:
You could use this one here: https://github.com/elastic/package-registry/blob/d4a6d00d7eed510b8630273acdf061ec69a47b58/util/package.go#L38

Reply:
No, I couldn't. The `Validate()` method is called by `Marshal()`, so the mentioned field is not filled yet.